beacon-chain/core/gloas/bid.go

```diff
@@ -0,0 +1,256 @@
+	"github.com/OffchainLabs/prysm/v7/beacon-chain/core/signing"
+	"github.com/OffchainLabs/prysm/v7/beacon-chain/state"
+	"github.com/OffchainLabs/prysm/v7/config/params"
+	"github.com/OffchainLabs/prysm/v7/consensus-types/blocks"
+	"github.com/OffchainLabs/prysm/v7/consensus-types/interfaces"
+	"github.com/OffchainLabs/prysm/v7/consensus-types/primitives"
+	"github.com/OffchainLabs/prysm/v7/crypto/bls"
+	"github.com/OffchainLabs/prysm/v7/crypto/bls/common"
+	ethpb "github.com/OffchainLabs/prysm/v7/proto/prysm/v1alpha1"
+	"github.com/OffchainLabs/prysm/v7/time/slots"
+)
+
+// ProcessExecutionPayloadBid processes a signed execution payload bid in the Gloas fork.
+// Spec v1.6.1 (pseudocode):
+// process_execution_payload_bid(state: BeaconState, block: BeaconBlock):
+//
+//	signed_bid = block.body.signed_execution_payload_bid
+//	bid = signed_bid.message
+//	builder_index = bid.builder_index
+//	builder = state.validators[builder_index]
+//	amount = bid.value
+//	if builder_index == block.proposer_index:
+//	  assert amount == 0
+//	  assert signed_bid.signature == G2_POINT_AT_INFINITY
+//	else:
+//	  assert has_builder_withdrawal_credential(builder)
+//	  assert verify_execution_payload_bid_signature(state, signed_bid)
+//	assert is_active_validator(builder, get_current_epoch(state))
+//	assert not builder.slashed
+//	assert amount == 0 or state.balances[builder_index] >= amount + pending_payments + pending_withdrawals + MIN_ACTIVATION_BALANCE
+//	assert bid.slot == block.slot
+//	assert bid.parent_block_hash == state.latest_block_hash
+//	assert bid.parent_block_root == block.parent_root
+//	assert bid.prev_randao == get_randao_mix(state, get_current_epoch(state))
+//	if amount > 0:
+//	  state.builder_pending_payments[...] = BuilderPendingPayment(weight=0, withdrawal=BuilderPendingWithdrawal(fee_recipient=bid.fee_recipient, amount=amount, builder_index=builder_index, withdrawable_epoch=FAR_FUTURE_EPOCH))
+//	state.latest_execution_payload_bid = bid
+func ProcessExecutionPayloadBid(st state.BeaconState, block interfaces.ReadOnlyBeaconBlock) error {
+	signedBid, err := block.Body().SignedExecutionPayloadBid()
+	if err != nil {
+		return fmt.Errorf("failed to get signed execution payload bid: %w", err)
+	}
+
+	wrappedBid, err := blocks.WrappedROSignedExecutionPayloadBid(signedBid)
+	if err != nil {
+		return fmt.Errorf("failed to wrap signed bid: %w", err)
+	}
+
+	bid, err := wrappedBid.Bid()
+	if err != nil {
+		return fmt.Errorf("failed to get bid from wrapped bid: %w", err)
+	}
+
+	builderIndex := bid.BuilderIndex()
+	proposerIndex := block.ProposerIndex()
+	amount := bid.Value()
+
+	builder, err := st.ValidatorAtIndex(builderIndex)
+	if err != nil {
+		return fmt.Errorf("failed to get builder validator: %w", err)
+	}
+
+	if builderIndex == proposerIndex {
+		if amount != 0 {
+			return fmt.Errorf("self-build amount must be zero, got %d", amount)
+		}
+		if wrappedBid.Signature() != common.InfiniteSignature {
```

Comment:
I think this should be `!bytes.Equal(wrappedBid.Signature(),common.InfiniteSignature)`